common/src/main/java/com/microsoft/identity/common/internal/numberMatch/NumberMatchHelper.kt

```diff
@@ -42,7 +43,16 @@ class NumberMatchHelper {
     // long enough for AuthApp to call the broker api to fetch the number match
     companion object {
         val TAG = NumberMatchHelper::class.java.simpleName
-        val numberMatchMap: HashMap<String, String> = HashMap()
+        val numberMatchMap: ConcurrentHashMap<String, String> = ConcurrentHashMap()
+
+        // Regex for GUID: 8-4-4-4-12 hex digits, case-insensitive
+        val guidRegex = Regex("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$")
+
+        // Regex for 8-character alphanumeric string
+        val alphaNum8Regex = Regex("^[A-Za-z0-9]{8}$")
+
+        // Regex for 2-digit number string
+        val twoDigitRegex = Regex("^\\d{2}$")
```

Comment:
I think a valid number can be single or double digits.

Reply:
I'm not sure, Say the number match is "05", would they send 05 or 5 as the string value? Let's check with arnav